Flat NoseCone geometries may calculate Negative Volume and Inertia

Reported on mailing list:
```
---------- Exception stack trace ----------
java.lang.reflect.InvocationTargetException
	at java.awt.EventQueue.invokeAndWait(Unknown Source)
	at java.awt.EventQueue.invokeAndWait(Unknown Source)
	at javax.swing.SwingUtilities.invokeAndWait(Unknown Source)
	at net.sf.openrocket.startup.SwingStartup.main(SwingStartup.java:76)
Caused by: net.sf.openrocket.util.BugException: BUG:   attempted to initialize an InertiaMatrix with a negative inertia value.
	at net.sf.openrocket.masscalc.RigidBody.<init>(RigidBody.java:24)
	at net.sf.openrocket.masscalc.MassCalculation.calculateAssembly(MassCalculation.java:279)
	at net.sf.openrocket.masscalc.MassCalculation.calculateAssembly(MassCalculation.java:310)
	at net.sf.openrocket.masscalc.MassCalculation.calculateAssembly(MassCalculation.java:310)
	at net.sf.openrocket.masscalc.MassCalculator.calculate(MassCalculator.java:105)
        . . .
```

--- a/core/src/net/sf/openrocket/rocketcomponent/SymmetricComponent.java
+++ b/core/src/net/sf/openrocket/rocketcomponent/SymmetricComponent.java
@@ -273,7 +273,7 @@ public abstract class SymmetricComponent extends BodyComponent implements Radial
 	@Override
 	public double getRotationalUnitInertia() {
 		if (rotationalInertia < 0) {
-			if (getComponentVolume() > 0.0000001)
+			if (getComponentVolume() > 0.0000001) // == 0.1cm^3
 				integrateInertiaVolume();
 			else
 				integrateInertiaSurface();
@@ -425,19 +425,18 @@ public abstract class SymmetricComponent extends BodyComponent implements Radial
 			final double inner;
 			final double dV;
 			
-			if (filled || r1 < thickness || r2 < thickness) {
+			final double hyp = MathUtil.hypot(r2 - r1, l);
+			final double height = thickness * hyp / l;
+			if (filled || r1 < height || r2 < height ) {
 				inner = 0;
 				dV = pil3 * (r1 * r1 + r1 * r2 + r2 * r2);
 			} else {
-				final double hyp = MathUtil.hypot(r2 - r1, l);
-				final double height = thickness * hyp / l;
 				dV = pil * height * (r1 + r2 - height);
-				inner = Math.max(outer - height, 0);
+				inner = Math.max(outer - height, 0.);
 			}
 			
 			rotationalInertia += dV * (pow2(outer) + pow2(inner)) / 2;
-			longitudinalInertia += dV * ((3 * (pow2(outer) + pow2(inner)) + pow2(l)) / 12
-					+ pow2(x + l / 2));
+			longitudinalInertia += dV * ((3 * (pow2(outer) + pow2(inner)) + pow2(l)) / 12 + pow2(x + l / 2));
 			
 			vol += dV;
 			

